### DISCUSSION TITLE
Nested tokens by category?

### DISCUSSION THREAD

**Discussion by davatron5000** [six7/figma-tokens#481] [Q&A] [answered]
Hey! Is it possible to have the nested category headings in the output JSON? e.g. Colors, Typography, etc...

How it currently looks:

```json
{
  "Brand Colors": {
    "Red": {
      "value": "#FF0000",
      "type": "color"
    },
    "Green": {
      "value": "#00FF00",
      "type": "color"
    },
    "Headings": {
      "H1 Headline": { ... },
      "H2 Headline": { ... }
    }
}
```

Ideal output:

```json
{
  "color": {
  "Brand Colors": {
    "Red": {
      "value": "#FF0000",
      "type": "color"
    },
    "Green": {
      "value": "#00FF00",
      "type": "color"
    }
  },
  "typography": {
    "Headings": {
      "H1 Headline": { ... },
      "H2 Headline": { ... }
    }
  }
}
```

Thanks!

**Comment by six7** [comment, marked as answer]
You could just include these as part of the name, I realize that makes those category headings appear twice in the UI, but other than that this would be fine. We're revamping how these categories are displayed in the future and one of the ideas would be to not group by type but rather just group by your naming structure.

So in your case just changing the JSON to

```
"color": {
  "Red": {
      "value": "#FF0000",
      "type": "color"
    },
    "Green": {
      "value": "#00FF00",
      "type": "color"
    },
},
"typography": {
    "Headings": {
      "H1 Headline": { ... },
      "H2 Headline": { ... }
    }
}
```

would do the trick.

**Reply by davatron5000** [reply]
Awesome. Thanks for the reply. I think custom grouping would be a cool feature that would work with how we're using tokens. Any timeframe on that? Or should I just check back?

**Reply by six7** [reply]
No timeframe yet, still collecting feedback around this. One drawback to grouping by name is that each token could be applied to different things, which breaks expectations (when using the plugin to Apply).

Maybe that's not that big of a deal though.

I'll keep you posted! 👍
